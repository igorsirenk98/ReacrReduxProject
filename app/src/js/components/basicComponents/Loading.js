import React from 'react';
import ReactLoading from 'react-loading';
 
const Loading = () => (
    <ReactLoading className="loading" type={'spinningBubbles'} color={'#000'} height={'5%'} width={'5%'} />
);
 
export default Loading;