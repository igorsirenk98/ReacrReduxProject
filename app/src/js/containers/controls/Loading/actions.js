import { LOADING_START, LOADING_END } from './action-types';

export const loadingStart = () => ({
	type: LOADING_START,
	payload: {
		isLoading: true
	},
});

export const loadingEnd = () => ({
	type: LOADING_END,
	payload: {
		isLoading: false
	},
});
