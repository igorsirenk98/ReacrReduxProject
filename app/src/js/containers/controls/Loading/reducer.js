import { LOADING_START, LOADING_END } from './action-types';

const initialState = {
	payload: {
		isLoading: false
	}
};

const loading = (state = initialState, action) => {
	switch (action.type) {
		case LOADING_START: {
			return {
				...state,
				payload: action.payload
			};
        }
        
        case LOADING_END: {
			return {
				...state,
				payload: action.payload
			};
		}
		default: {
			return state;
		}
	}
};

export default loading;
