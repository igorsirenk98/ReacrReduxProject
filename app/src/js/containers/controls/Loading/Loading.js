import React, { Component } from 'react';
import ReactLoading from 'react-loading';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { loadingStart, loadingEnd } from './actions';

const mapStateToProps = state => ({
    isLoading: state.loadingAnimation.payload
});

const mapDispatchToProps = dispatch => {
    return bindActionCreators({
        loadingStart,
        loadingEnd
    }, dispatch);
};

export class Loading extends Component {
    render() {
        return (
            <ReactLoading
                type="spinningBubbles"
                color="#000"
                height="50px"
                width="50px"
            />
        )
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Loading);