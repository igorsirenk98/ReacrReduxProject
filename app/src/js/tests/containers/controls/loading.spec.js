import React from 'react';
import { shallow } from 'enzyme';
import { Loading } from '../../../containers/controls/Loading/Loading';

describe('Loading component', () => {
    const props = {
        isLoading: false,
        loadingStart: () => { },
        loadingEnd: () => { },
    }

    describe('Loading initial', () => {
        const loading = shallow(<Loading {...props}/>);

        it('renders correctly', () => {
            expect(loading.debug()).toMatchSnapshot()
        })
    });

    describe('Loading start', () => {
        const nextProps = {
            ...props,
            isLoading: true,
        }

        const loading = shallow(<Loading {...nextProps}/>);

        it('renders correctly', () => {
            expect(loading.debug()).toMatchSnapshot()
        })
    });
})
