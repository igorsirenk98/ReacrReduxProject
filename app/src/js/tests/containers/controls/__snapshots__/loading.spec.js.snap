// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component Loading initial renders correctly 1`] = `"<Loading type=\\"spinningBubbles\\" color=\\"#000\\" height=\\"50px\\" width=\\"50px\\" delay={0} />"`;

exports[`Loading component Loading start renders correctly 1`] = `"<Loading type=\\"spinningBubbles\\" color=\\"#000\\" height=\\"50px\\" width=\\"50px\\" delay={0} />"`;
